### PR TITLE
chore: migrate eslint to flat config

### PR DESCRIPTION
**Pull request by pnodet**

### CODE REVIEW

**Review by tim-smart — APPROVED** [review]
Look good to me. I'll let @fubhy our tooling guru weigh in before merging.

**Comment by fubhy** [comment]
Thanks @pnodet, this is great.

If I remember correctly, this was the only missing piece before we can upgrade eslint to v9. I've added a commit to try exactly that. Let's see if it works and then it should be ready to go!

**Pull request merged**
